src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,159 @@
-                    continue
-                if x1 < frame.shape[1]/2 and x2 < frame.shape[1]/2:
-                    # <-- If the slope is negative, left group
-                    # x_array_left.append([x1,x2])
-                    # y_array_left.append([y1,y2])
-                    cv2.line(crop2, (x1, y1), (x2, y2), (0, 0, 255), 3)
-                    # cv2.circle(crop, (x1, y1),(5), (0,0,255), 3)
-                    # cv2.circle(crop, (x2, y2),(5), (0,0,255), 3)
-                else:  # <-- Otherwise, right group.
-                    # x_array_right.append([x1,x2])
-                    # y_array_right.append([y1,y2])
-                    cv2.line(crop2, (x1, y1), (x2, y2), (0, 255, 0), 3)
-                    # cv2.circle(crop, (x1, y1),(5), (0,255,0), 3)
-                    # cv2.circle(crop, (x2, y2),(5), (0,255,0), 3)
-            circles = crop
-            lines = crop2
-
-    if linesP is None or len(linesP) < 40:
-        skipped_frames += 1
-
-    cv2.imshow('Crop', crop)
-    # linesAndCircles = cv2.addWeighted(lines, 0.5, circles, 0.5,0)
-    # final = cv2.addWeighted(processed, 0.8, linesAndCircles, 0.5,0)
-    # cv2.imshow('Original', frame)
-    # cv2.imshow('Combo', final)
-    # print(skipped_frames)
-
-    # FIXME: what is this crap, please FIX ME
-    # plt.plot(np.unique(x_array_right),
-    #          np.poly1d(np.polyfit(x_array_right, y_array_right, 1))
-    #          (np.unique(x_array_right)))
-
-    # right_line_slope,right_line_intercept = best_fit(np.asarray(x_array_right),np.asarray(y_array_right))
-    # left_line_slope,left_line_intercept = best_fit(np.asarray(x_array_left),np.asarray(y_array_left))
-
-    toc = time.time()
-    delta_t = toc - tic
-    print("FPS: {}".format(1/delta_t))
-
-    key = cv2.waitKey(25)
-    if key == 27:
-        break
-
-video.release()
-cv2.destroyAllWindows
+
+# Numpy and OpenCV
+import cv2
+import numpy as np
+
+# ROS imports
+import rospy
+from sensor_msgs.msg import Image  # ROS Image Message
+from cv_bridge import CvBridge, CvBridgeError  # Converts b/w OpenCV Image and ROS Image Message
+from magellan_core.msg import Float64Arr  # ROS Float64Arr type msg
+
+# Global Constant
+VERBOSE = True
+
+
+# Class PubSubNode: Subscribes to the /camera/color/image_raw topic(RGB Images from realsense camera) and publishes
+#                   to the /perception/color/image_processed topic(Processed RGB raw Images) and to the
+#                   /perception/detectedPoints topic(Detected Points as a Float64Arr msg)
+# TODO: Publish to other topics: Depth image; etc..
+class PubSubNode:
+    def __init__(self):
+        '''Initialize ros publisher, ros subscriber'''
+        self._lines_object = Lines()
+        # topic where we publish
+        self._image_pub = rospy.Publisher("/perception/color/image_processed", Image, queue_size=5)
+        self._point_arr_pub = rospy.Publisher('/perception/detectedPoints', Float64Arr, queue_size=10)
```

Comment:
Can you set the topic name to something like `detected_points`? It's not common to use camel case for topic names.

Reply:
fixed